jmeter/src/main/java/org/eclipse/hono/jmeter/ui/HonoCommanderSamplerUI.java

```diff
@@ -0,0 +1,156 @@
+        sampler.configureServerOptions(honoServerOptions);
+        tenant.setText(sampler.getTenant());
+        commandTimeOut.setText(sampler.getCommandTimeout());
+        command.setText(sampler.getCommand());
+        commandPayload.setText(sampler.getCommandPayload());
+        protocol.setSelectedItem(format(protocolDisplayFormat, sampler.getProtocol()));
+        protocolDescription.setText(getDescriptionTextForSelectedProtocol());
+
+    }
+
+    @Override
+    public void clearGui() {
+        super.clearGui();
+        honoServerOptions.clearGui();
+        tenant.setText("");
+        commandTimeOut.setText("");
+        command.setText("");
+        commandPayload.setText("");
+        protocol.setSelectedItem(format(protocolDisplayFormat, protocols[0]));
+        protocolDescription.setText(getDescriptionTextForSelectedProtocol());
+    }
+
+    private JPanel getProtocolPanel() {
+        final JPanel protocolOuterPanel = new VerticalPanel(50, 0.0F);
+        protocolOuterPanel.setBorder(
+                BorderFactory.createTitledBorder(BorderFactory.createEtchedBorder(),
+                        "Protocol used for communication between devices and hono"));
+        final JPanel protocolPanel = new JPanel(new BorderLayout(10, 50));
+        setProtocolDescriptionTextAreaDesign(protocolPanel);
+        protocolPanel.add(protocol, BorderLayout.LINE_START);
+        protocolPanel.add(protocolDescription, BorderLayout.CENTER);
+        protocol
+                .addActionListener(e -> protocolDescription.setText(getDescriptionTextForSelectedProtocol()));
+        protocolOuterPanel.add(protocolPanel);
+        return protocolOuterPanel;
+    }
+
+    private void setProtocolDescriptionTextAreaDesign(final JPanel parentPanel) {
+        protocolDescription
+                .setSize(parentPanel.getWidth() - protocol.getWidth() - 20, 10);
+        protocolDescription.setLineWrap(true);
+        protocolDescription.setWrapStyleWord(true);
+        protocolDescription.setEditable(false);
+        protocolDescription.setMargin(new Insets(15, 15, 15, 15));
+        protocolDescription.setBackground(parentPanel.getBackground());
+        protocolDescription.setBorder(BorderFactory.createLineBorder(Color.LIGHT_GRAY, 1));
+    }
+
+    private String getDescriptionTextForSelectedProtocol() {
+        final String prototypeSelected = ((String) protocol.getSelectedItem()).trim();
+        if (prototypeSelected.equals("http")) {
+            return "Devices use Http event/telemetry messages to inform their command readiness by setting hono-ttd value. In this case, commands are triggered by this sampler on arrival of those messages.";
+        } else if (prototypeSelected.equals("mqtt")) {
+            return "Devices subscribe to a specified mqtt topic to inform their command readiness. In this case, this sampler send one command per sample to the devices.";
+        }
+        return "No description available.";
+    }
```

Comment:
In general I think it would be better not to distinguish between MQTT/HTTP, but between the modes of devices with permanent connection (e.g. MQTT, AMQP) and devices that occasionally report (e.g. with HTTP calls).

The whole point of Hono is that the protocol doesn't really matter. However, the basic type of connection of the Devices may already play a role for the case.

Reply:
I think with your MQTT example it is the opposite. If a device makes a MQTT connection or subscription only for a short time, the subscription (ttd event) should be the trigger and it need to be used the same way as the HTTP right now.

As I understand it, the technical difference is whether you can assume that the device is completely accessible for commands during the test or the device is only accessible if it has sent a ttd event. How this maps to the protocols should not matter from the point of view. For example, there could be an HTTP adapter that supports sockets. Or just that by MQTT is always only briefly connected. You would also want to test such a scenario via JMeter. For example, to check the costs of MQTT connection setups.

I am not sure what you mean with uni/bidirectional as the commands in the moment are always bidirectional FMPOV, since we always expect a command response to be sure, the command arrived. In the future we will maybe have unidirectional calls. (#669, #808)

Reply:
1. As we discussed, instead of grouping based on protocols, I group now based on triggers (initiated by device or sampler) as in the attached screenshot. WDYT?

![image](https://user-images.githubusercontent.com/18659982/45090234-11705380-b10f-11e8-8fa1-e35bd92791d8.png)

2. As you noticed, there is a chance that after sending commands and before the devices receive them, devices can unsubscribe. In this case, it is recorded as error. It can be overcome, by  checking if the device is still subscribed, before counting it as an error.

Reply:
I like this much more then MQTT/HTTP, because it shows the cases from the application point of view (for which the protocol should not matter).